pkg/descheduler/descheduler_test.go

```diff
@@ -29,45 +27,120 @@ import (
 }
 
-// Convert continues a conversion.
-func (s scope) Convert(src, dest interface{}) error {
-	return s.converter.Convert(src, dest, s.meta)
+func removePodsViolatingNodeTaintsPolicy() *api.DeschedulerPolicy {
+	return &api.DeschedulerPolicy{
+		Profiles: []api.DeschedulerProfile{
+			{
+				Name: "Profile",
+				PluginConfigs: []api.PluginConfig{
+					{
+						Name: "RemovePodsViolatingNodeTaints",
+						Args: &removepodsviolatingnodetaints.RemovePodsViolatingNodeTaintsArgs{},
+					},
+					{
+						Name: "DefaultEvictor",
+						Args: &defaultevictor.DefaultEvictorArgs{},
+					},
+				},
+				Plugins: api.Plugins{
+					Filter: api.PluginSet{
+						Enabled: []string{
+							"DefaultEvictor",
+						},
+					},
+					Deschedule: api.PluginSet{
+						Enabled: []string{
+							"RemovePodsViolatingNodeTaints",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func removeDuplicatesPolicy() *api.DeschedulerPolicy {
+	return &api.DeschedulerPolicy{
+		Profiles: []api.DeschedulerProfile{
+			{
+				Name: "Profile",
+				PluginConfigs: []api.PluginConfig{
+					{
+						Name: "RemoveDuplicates",
+						Args: &removeduplicates.RemoveDuplicatesArgs{},
+					},
+					{
+						Name: "DefaultEvictor",
+						Args: &defaultevictor.DefaultEvictorArgs{},
+					},
+				},
+				Plugins: api.Plugins{
+					Filter: api.PluginSet{
+						Enabled: []string{
+							"DefaultEvictor",
+						},
+					},
+					Balance: api.PluginSet{
+						Enabled: []string{
+							"RemoveDuplicates",
+						},
+					},
+				},
+			},
+		},
+	}
 }
 
-// Meta returns the meta object that was originally passed to Convert.
-func (s scope) Meta() *conversion.Meta {
-	return s.meta
+func initDescheduler(t *testing.T, ctx context.Context, internalDeschedulerPolicy *api.DeschedulerPolicy, objects ...runtime.Object) (*options.DeschedulerServer, *descheduler, *fakeclientset.Clientset, func()) {
+	client := fakeclientset.NewSimpleClientset(objects...)
+	eventClient := fakeclientset.NewSimpleClientset(objects...)
+
+	rs, err := options.NewDeschedulerServer()
+	if err != nil {
+		t.Fatalf("Unable to initialize server: %v", err)
+	}
+	rs.Client = client
+	rs.EventClient = eventClient
+
+	sharedInformerFactory := informers.NewSharedInformerFactoryWithOptions(rs.Client, 0, informers.WithTransform(trimManagedFields))
+	eventBroadcaster, eventRecorder := utils.GetRecorderAndBroadcaster(ctx, client)
+
+	descheduler, err := newDescheduler(rs, internalDeschedulerPolicy, "v1", eventRecorder, sharedInformerFactory)
+	if err != nil {
+		eventBroadcaster.Shutdown()
+		t.Fatalf("Unable to create a descheduler instance: %v", err)
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
returning cancel is an odd signature, perhaps pass in the cancel context?

Reply:
The function still needs to return the event broadcaster shutdown. Are you suggesting to find a different way of canceling the context and calling the shutdown?

Reply:
ah true. we can leave it as-is then

Reply:
could create the ctx with cancel at a higher level and pass it in here? 
the broadcaster should shutdown on ctx cancelation?
```StartRecordingToSink(stopCh <-chan struct{})```

Reply:
That works, thx :) https://github.com/kubernetes-sigs/descheduler/pull/1460